linera-sdk/src/views/system_api.rs

```diff
@@ -346,3 +346,39 @@ impl From<KeyValueStore> for ViewStorageContext {
         }
     }
 }
+
+#[cfg(with_testing)]
+#[tokio::test]
+async fn test_key_value_store_mock() -> anyhow::Result<()> {
+    // Create a mock key-value store for testing
+    let store = KeyValueStore::mock();
+    let mock_store = store.to_mut();
+
+    // Check if key exists
+    let is_key_existing = mock_store.contains_key(b"foo").await?;
+    assert!(!is_key_existing);
+
+    // Check if keys exist
+    let is_keys_existing = mock_store
+        .contains_keys(vec![b"foo".to_vec(), b"bar".to_vec()])
+        .await?;
+    assert!(!is_keys_existing[0]);
+    assert!(!is_keys_existing[1]);
+
+    // Read and write values
+    let mut batch = Batch::new();
+    batch.put_key_value(b"foo".to_vec(), &32_u128)?;
+    batch.put_key_value(b"bar".to_vec(), &42_u128)?;
+    mock_store.write_batch(batch).await?;
+
+    let is_key_existing = mock_store.contains_key(b"foo").await?;
+    assert!(is_key_existing);
+
+    let value = mock_store.read_value(b"foo").await?;
+    assert_eq!(value, Some(32_u128));
+
+    let value = mock_store.read_value(b"bar").await?;
+    assert_eq!(value, Some(42_u128));
+
+    Ok(())
+}
```

Comment:
Two small comments:

1. We usually place the tests in a separate module
2. `with_testing` would build the test even when it doesn't need to run, so it's best to use `#[cfg(test)]` instead

```suggestion
#[cfg(all(test, not(target_arch = "wasm32")))]
mod tests {
    use super::*;

    #[tokio::test]
    async fn test_key_value_store_mock() -> anyhow::Result<()> {
        // Create a mock key-value store for testing
        let store = KeyValueStore::mock();
        let mock_store = store.to_mut();

        // Check if key exists
        let is_key_existing = mock_store.contains_key(b"foo").await?;
        assert!(!is_key_existing);

        // Check if keys exist
        let is_keys_existing = mock_store
            .contains_keys(vec![b"foo".to_vec(), b"bar".to_vec()])
            .await?;
        assert!(!is_keys_existing[0]);
        assert!(!is_keys_existing[1]);

        // Read and write values
        let mut batch = Batch::new();
        batch.put_key_value(b"foo".to_vec(), &32_u128)?;
        batch.put_key_value(b"bar".to_vec(), &42_u128)?;
        mock_store.write_batch(batch).await?;

        let is_key_existing = mock_store.contains_key(b"foo").await?;
        assert!(is_key_existing);

        let value = mock_store.read_value(b"foo").await?;
        assert_eq!(value, Some(32_u128));

        let value = mock_store.read_value(b"bar").await?;
        assert_eq!(value, Some(42_u128));

        Ok(())
    }
}
```

Reply:
Thx for help! @jvff 
Done e09f125